Fixed the search icon color in the navigation header

jdatapple opened the pull request:
#### b770db0515fe568a19130be241b61b08359e9542
<pre>
Fixed the search icon color in the navigation header
<a href="https://bugs.webkit.org/show_bug.cgi?id=246105">https://bugs.webkit.org/show_bug.cgi?id=246105</a>

Reviewed by Devin Rousso.

* Websites/webkit.org/wp-content/themes/webkit/header.php:
* Websites/webkit.org/wp-content/themes/webkit/style.css:
(:root):
(#site-nav .search-input):
(@media only screen and (max-width: 1015px) #site-nav .search-input,):
(@media only screen and (max-width: 1015px) .search-input): Deleted.

Canonical link: <a href="https://commits.webkit.org/255199@main">https://commits.webkit.org/255199@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/cbf5458dab633e406ab60c0f35e612a10cf47608

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/6/builds/91689 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/77/builds/919 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/43/builds/22287 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/8/builds/101384 "Built successfully") | 
| | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/76/builds/918 "Built successfully") | [  ~~🛠 mac-debug~~](https://ews-build.webkit.org/#/builders/71/builds/29508 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/36/builds/84016 "Built successfully") | [  ~~🛠 wincairo~~](https://ews-build.webkit.org/#/builders/12/builds/97709 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/19/builds/97347 "Passed tests") | [  ~~🧪 ios-wk2~~](https://ews-build.webkit.org/#/builders/78/builds/542 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🛠 mac-AS-debug~~](https://ews-build.webkit.org/#/builders/61/builds/78332 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 gtk-wk2~~](https://ews-build.webkit.org/#/builders/35/builds/27500 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | 
| | [  ~~🧪 api-ios~~](https://ews-build.webkit.org/#/builders/9/builds/82466 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 api-mac~~](https://ews-build.webkit.org/#/builders/3/builds/82100 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 api-gtk~~](https://ews-build.webkit.org/#/builders/34/builds/70538 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/66/builds/35809 "Built successfully") | [  ~~🧪 mac-wk1~~](https://ews-build.webkit.org/#/builders/73/builds/16127 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/67/builds/33563 "Built successfully") | [  ~~🧪 mac-wk2~~](https://ews-build.webkit.org/#/builders/70/builds/17224 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/63/builds/37407 "Built successfully") | [  ~~🧪 mac-AS-debug-wk2~~](https://ews-build.webkit.org/#/builders/61/builds/78332 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | 
| [✅ 🛠 🧪 unsafe-merge](https://ews-build.webkit.org/#/builders/75/builds/1618 "Built successfully and passed tests") | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/65/builds/39312 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/62/builds/36341 "Passed tests") | | 
<!--EWS-Status-Bubble-End-->

karlcow reviewed (approved):
lgtm

dcrousso commented on Websites/webkit.org/wp-content/themes/webkit/style.css:
```diff
@@ -2549,6 +2554,7 @@ p .search-term {
         padding: 0 3rem 3rem 0;
     }
 
+    #site-nav .search-input,
     .search-input {
```
why do we need to add the `#site-nav`?  seems unnecessary

jdatapple replied:
There is a generic definition of `.search-input` used everywhere as the search input can be used in dark and light parts of the page regardless of dark and light mode preferences. 

The issue this patch fixes only occurs in the site navigation in the header, but only when the header uses desktop layout. At smaller viewport sizes the menu design changes and makes use of the originally defined default. This contextual override is necessary.

I believe it is scoped appropriately.

jdatapple replied:
This specific override, resets the design to back to the defaults in smaller viewport layouts.

dcrousso commented on Websites/webkit.org/wp-content/themes/webkit/style.css:
```diff
@@ -42,6 +42,7 @@ Version: 1.0
     --submit-button-text-color: hsl(0, 0%, 100%);
     
     --search-glyph: url('images/search.svg#dark');
+    --search-glyph-light: url('images/search.svg#light');
```
should we maybe give this a different name (or even just hardcode it for the one use case) so it doesn't confuse future developers who wonder "why do we have this when `--search-glyph` has the same value in dark mode?

jdatapple replied:
`--search-glyph` behavior changes based on color scheme. `--search-glyph-light` stays light in both light and dark mode, which is exactly what is needed to specify in the unique desktop layout header case (it should always use "light"). 

I thought about defining it as `--search-glyph-inverse`, but given that there was no need to for different setting in dark mode, it would have no useful meaning. `--search-glyph-light` seems the most apt term for what this custom property is defined as.

dcrousso commented on Websites/webkit.org/wp-content/themes/webkit/style.css:
```diff
@@ -2297,6 +2298,10 @@ header .menu-item { /* add bottom dimension to main menu items */
     transition: 200ms ease-out width, 200ms ease-in background-color;
 }
 
+#site-nav .search-input {
```
`#site-nav` only appears to be used in one other spot.  is this the best selector to use?

jdatapple replied:
Yes, it addresses a singularly specific case, exactly where an id-selector is appropriate. We're not defining a class of behaviors, but a very specific and scoped behavior.

jdatapple committed:
Fixed the search icon color in the navigation header
https://bugs.webkit.org/show_bug.cgi?id=246105

Reviewed by Devin Rousso.

* Websites/webkit.org/wp-content/themes/webkit/header.php:
* Websites/webkit.org/wp-content/themes/webkit/style.css:
(:root):
(#site-nav .search-input):
(@media only screen and (max-width: 1015px) #site-nav .search-input,):
(@media only screen and (max-width: 1015px) .search-input): Deleted.

Canonical link: https://commits.webkit.org/255199@main
